### ISSUE TITLE
[AKS] Nodepool management misbehaves if initial nodepool is named 'pool1'

### ISSUE THREAD

**Issue by valaparthvi** [rancher/dashboard#11406]
**Setup**
- Rancher version: v2.9-a9355940c0629ca419b7dd3e4098c4ea0e52c0c0-head
- Rancher UI Extensions: master [46a44c1](https://github.com/rancher/dashboard/commit/46a44c11ddbb14581db9793aadc3512e1362edf1)
- Browser type & version:

**Describe the bug**

If the cluster initially has a single nodepool named 'pool1', subsequently adding/removing another nodepool becomes confusing.

**To Reproduce**

1. Create an AKS cluster with only nodepool 'pool1'.
2. Once the cluster has provisioned, try adding a new nodepool. 

**Result**
Nodepool with name 'pool1' will be re-added. If you try removing the nodepool, all nodepools are removed.

**Expected Result**

It should not misbehave when trying to add a new nodepool
**Screenshots**

[Screencast from 07-10-2024 05:37:30 PM.webm](https://github.com/rancher/dashboard/assets/11015077/aef37423-42a5-48f1-8959-e6de2b902fee)

Also happens when renaming `agentpool` to `pool1` during initial cluster creation.
[Screencast from 07-10-2024 05:42:08 PM.webm](https://github.com/rancher/dashboard/assets/11015077/be396910-52b9-416b-9448-320301d694e1)

**Additional context**

Workaround:
1. Add another nodepool, pool2 with system mode and update.
2. Once the update is complete, edit the cluster to remove the nodepool 'pool1'.
3. Once this update is complete, nodepool management seems to be back to working as expected again.
4. When a new nodepool is added it will be called 'pool1', and the next one will be called 'pool3'.

**Comment by valaparthvi** [comment]
@richard-cox @mantis-toboggan-md Can this get some love?

**Comment by valaparthvi** [comment]
Tested on Rancher: v2.9-81337b95660cba868629d6fe769bd7e8242b5aee-head, Dashboard: master 3da1205.

_It is only partially fixed_.

I would not it is blocking anymore, but the behavior can still be improvised.

[Screencast from 07-17-2024 06:05:04 PM.webm](https://github.com/user-attachments/assets/75ec9ff5-9132-4863-82b4-ef6dc43f4c4a)

**Comment by mantis-toboggan-md** [comment]
I thought it could be better as well @valaparthvi but I didn't want to add too much to the 2.9.0 release at this stage so I filed an issue to improve node pool/group naming across the 3 hosted providers https://github.com/rancher/dashboard/issues/11451

**Comment by valaparthvi** [comment]
@mantis-toboggan-md In that case, feel free to close this issue and track this in the one you filed.

**Comment by izaac** [comment]
Tech notes and test coverage details in PR. A manual QA validation will be executed before closing.

**Comment by valaparthvi** [comment]
I am closing this issue in favor of tracking further enhancements in #11451.

**Comment by valaparthvi** [comment]
> Tech notes and test coverage details in PR. A manual QA validation will be executed before closing.

Manual QA validation is already done, see https://github.com/rancher/dashboard/issues/11406#issuecomment-2233221924.

**Comment by izaac** [comment]
I missed that comment. please do as needed thanks
